cdap-watchdog/src/main/java/co/cask/cdap/logging/save/CheckpointManager.java

```diff
@@ -16,125 +16,26 @@
-      } else {
-        throw new RuntimeException(String.format("Table %s is not TransactionAware, " +
-                                                   "Exception while trying to cast it to TransactionAware. " +
-                                                   "Please check why the table is not TransactionAware", table));
-      }
-    } catch (Exception e) {
-      throw new RuntimeException(String.format("Error accessing %s table", tableUtil.getMetaTableName()), e);
-    }
-  }
-
-  public void saveCheckpoint(final Map<Integer, Checkpoint> checkpoints) throws Exception {
-    // if the checkpoints have not changed, we skip writing to table and return.
-    if (lastCheckpoint.equals(checkpoints)) {
-      return;
-    }
-
-    execute(new TransactionExecutor.Procedure<Table>() {
-      @Override
-      public void apply(Table table) throws Exception {
-        for (Map.Entry<Integer, Checkpoint> entry : checkpoints.entrySet()) {
-          byte[] key = Bytes.add(rowKeyPrefix, Bytes.toBytes(entry.getKey()));
-          Checkpoint checkpoint = entry.getValue();
-          table.put(key, OFFSET_COLNAME, Bytes.toBytes(checkpoint.getNextOffset()));
-          table.put(key, MAX_TIME_COLNAME, Bytes.toBytes(checkpoint.getMaxEventTime()));
-        }
-        // update last checkpoint
-        lastCheckpoint = ImmutableMap.copyOf(checkpoints);
-      }
-    });
-    LOG.trace("Saving checkpoints for partitions {}", checkpoints);
-  }
-
-  public Map<Integer, Checkpoint> getCheckpoint(final Set<Integer> partitions) throws Exception {
-    return execute(new TransactionExecutor.Function<Table, Map<Integer, Checkpoint>>() {
-      @Override
-      public Map<Integer, Checkpoint> apply(Table table) throws Exception {
-        Map<Integer, Checkpoint> checkpoints = new HashMap<Integer, Checkpoint>();
-        for (final int partition : partitions) {
-          Row result =
-            table.get(Bytes.add(rowKeyPrefix, Bytes.toBytes(partition)));
-          checkpoints.put(partition,
-                             new Checkpoint(result.getLong(OFFSET_COLNAME, -1),
-                                            result.getLong(MAX_TIME_COLNAME, -1)));
-        }
-        return checkpoints;
-      }
-    });
-  }
-
-  public Checkpoint getCheckpoint(final int partition) throws Exception {
-    Checkpoint checkpoint = execute(new TransactionExecutor.Function<Table, Checkpoint>() {
-      @Override
-      public Checkpoint apply(Table table) throws Exception {
-        Row result = table.get(Bytes.add(rowKeyPrefix, Bytes.toBytes(partition)));
-        return new Checkpoint(result.getLong(OFFSET_COLNAME, -1), result.getLong(MAX_TIME_COLNAME, -1));
-      }
-    });
-    LOG.trace("Read checkpoint {} for partition {}", checkpoint, partition);
-    return checkpoint;
-  }
+public interface CheckpointManager {
+
+  /**
+   * Persists the given map of {@link Checkpoint}s.
+   */
+  void saveCheckpoint(Map<Integer, Checkpoint> checkpoints) throws Exception;
```

Comment:
This is not new code. I just turn `CheckpointManager` into an interface and the original implementation is renamed to `DefaultCheckpointManager`. Anyway, changed the name.